share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,628 @@
+## Wake Tutorial
+
+### Invoking wake
+
+1. Create a wake database: `wake --init .`
+
+2. Run command for wake: `wake -vx '<function_name> <parameter_if_any>'`
+
+* If you want to display a string you can that using the below function by writing the string to be displayed inisde "" and call the function abc . Ex: wake -x 'abc'
+```
+def abc = "Hello World"
+```
+
+* The other way to do this is `wake -x '"Hello World"'`
+
+### Defining Functions
+A function may be defines using the keyword def. Simple function definition takes the form
+`def <fun name> <parameter> = <expression>`
+
+* Below function  double takes an argment x and gives the output by multiplying the x by 2
+```
+def double x = 2*x
+```
+
+* Below function triple takes an argument x and gives the output by multiplying the x by 3
+```
+def triple x = 3*x
+```
+
+* Function inc increments the values by 1 and returns the value
+```
+def inc x = x+1
+```
+
+* Function adda takes an argument of type string and concantenates it with b and gives the output
+```
+def adda s = "{s}b"
+```
+
+* Function times4 takes an argument x and applies the function double twice on x
+```
+def times4 x = double (double x)
+```
+
+* Function aveI takes 2 arguments x and y and adds two values and divide the result by 2. This function shows the other was of defining the parameters using lambda operator
```

Comment:
what is the lambda operator? This is the first time you mentioned it. How is this example showing it at all?

Reply:
what is the name `aveI` supposed to convey? This is taking an average, what is I?

Reply:
You can use lambda operator to create the parameters by defining then in the function body. so `def function a b = a+b` 
is same as `def function = \a \b (a+b)`

aveI conveys averageofInteger numbers